device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -358,7 +357,7 @@ public void sendReportedPropertiesAsync(ReportedPropertiesParameters reportedPro
      * @throws IOException if called when client is not opened or called before starting twin.
      * @throws IllegalArgumentException if reportedProperties is null or empty or if version is negative
      */
-    public void sendReportedPropertiesAsync(Set<Property> reportedProperties, Integer version, CorrelatingMessageCallback correlatingMessageCallback, Object correlatingMessageCallbackContext, IotHubEventCallback reportedPropertiesCallback, Object reportedPropertiesCallbackContext) throws IOException, IllegalArgumentException
+    private void sendReportedPropertiesAsync(Set<Property> reportedProperties, Integer version, CorrelatingMessageCallback correlatingMessageCallback, Object correlatingMessageCallbackContext, IotHubEventCallback reportedPropertiesCallback, Object reportedPropertiesCallbackContext) throws IOException, IllegalArgumentException
```

Comment:
This is a public method I made earlier to allow a customer to correlate a message all the way across the send/recv stack. Probably shouldn't remove this. Do I need to add a test for this?

Reply:
I'll revert this change. You're right that this should stay public. No need to add a test just for the sake of keeping the code analysis from flagging this as a method to reduce access level on